### PR TITLE
docs(firebase_auth): fix broken links

### LINKED ISSUE #3889: [Firebase_auth][📚] Links to all sign-in methods in Other sign-in methods is broken
Links to all the [other sign-in methods](https://firebase.flutter.dev/docs/auth/usage/#other-sign-in-methods) is broken.

Attaching one screenshot here.

broken links bug

### PR DESCRIPTION
**Pull request by sidrao2006**
## Description

Fix typos in #3889

## Related Issues

Fixes #3889

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by sidrao2006** [comment]
cc @Ehesp @Salakar

File: docs/auth/usage.mdx [hunk 1]
```diff
@@ -207,7 +207,7 @@ try {
 }
 ```
 
-To learn more about how you can handle any errors which are thrown from the method, view the [Error Handling](error-handling)
+To learn more about how you can handle any errors which are thrown from the method, view the [Error Handling](../error-handling)
```
Comment (Salakar) [inline docs/auth/usage.mdx:210]:
```suggestion
To learn more about how you can handle any errors which are thrown from the method, view the [Error Handling](error-handling)
```

File: docs/auth/usage.mdx [hunk 2]
```diff
@@ -141,7 +141,7 @@ The method is a two-step operation; it will first create the new account (if it
 then automatically sign in the user in to that account. If you are listening to changes in [authentication state](#authentication-state),
 a new event will be sent to your listeners.
 
-To learn more about how you can handle any errors which are thrown from the method, view the [Error Handling](error-handling)
+To learn more about how you can handle any errors which are thrown from the method, view the [Error Handling](../error-handling)
```
Comment (Salakar) [inline docs/auth/usage.mdx:144]:
```suggestion
To learn more about how you can handle any errors which are thrown from the method, view the [Error Handling](error-handling)
```

File: docs/auth/usage.mdx [hunk 3]
```diff
@@ -106,7 +106,7 @@ UserCredential userCredential = await FirebaseAuth.instance.signInAnonymously();
 Once successfully resolved, the user will be granted an anonymous account. If you are listening to changes in
 [authentication state](#authentication-state), a new event will be sent to your listeners.
 
-To learn more about how you can handle any errors which are thrown from the method, view the [Error Handling](error-handling)
+To learn more about how you can handle any errors which are thrown from the method, view the [Error Handling](../error-handling)
```
Comment (Salakar) [inline docs/auth/usage.mdx:109]:
```suggestion
To learn more about how you can handle any errors which are thrown from the method, view the [Error Handling](error-handling)
```

**Review by Salakar — APPROVED** [review]

**Pull request merged**